ddtrace/tracer/textmap_test.go

```diff
@@ -637,6 +637,152 @@ func TestEnvVars(t *testing.T) {
+					"w3cTraceID":   "00000000000000001111111111111111",
+					"_dd.p.dm":     "-2",
+					"_dd.p.usr.id": "baz64==",
+					"tracestate":   "dd=s:0;o:rum;t.dm:-2;t.usr.id:baz64~~,othervendor=t61rcWkgMzE"},
+			},
+			{
+				in: TextMapCarrier{
+					traceparentHeader: "00-00000000000000001111111111111111-2222222222222222-01",
+					tracestateHeader:  "dd=s:2;o:rum:rum;t.dm:-4;t.usr.id:baz64~~,othervendor=t61rcWkgMzE",
+				},
+				traceID:  1229782938247303441,
+				spanID:   2459565876494606882,
+				priority: 2, // tracestate priority takes precedence
+				origin:   "rum:rum",
+				propagatingTags: map[string]string{
+					"w3cTraceID":   "00000000000000001111111111111111",
+					"_dd.p.dm":     "-4",
+					"_dd.p.usr.id": "baz64==",
+					"tracestate":   "dd=s:2;o:rum:rum;t.dm:-4;t.usr.id:baz64~~,othervendor=t61rcWkgMzE",
+				},
+			},
+			{
+				in: TextMapCarrier{
+					traceparentHeader: "00-00000000000000001111111111111111-2222222222222222-01",
+					tracestateHeader:  "dd=s:;o:rum:rum;t.dm:-4;t.usr.id:baz64~~,othervendor=t61rcWkgMzE",
+				},
+				traceID:  1229782938247303441,
+				spanID:   2459565876494606882,
+				priority: 1, // traceparent priority takes precedence
+				origin:   "rum:rum",
+				propagatingTags: map[string]string{
+					"w3cTraceID":   "00000000000000001111111111111111",
+					"_dd.p.dm":     "-4",
+					"_dd.p.usr.id": "baz64==",
+					"tracestate":   "dd=s:;o:rum:rum;t.dm:-4;t.usr.id:baz64~~,othervendor=t61rcWkgMzE",
+				},
+			},
+			{
+				in: TextMapCarrier{
+					traceparentHeader: " \t-00-00000000000000001111111111111111-2222222222222222-01 \t-",
+					tracestateHeader:  "dd=s:;o:rum:rum;t.dm:-4;t.usr.id:baz64~~,othervendor=t61rcWkgMzE",
+				},
+				traceID:  1229782938247303441,
+				spanID:   2459565876494606882,
+				priority: 1, // traceparent priority takes precedence
+				origin:   "rum:rum",
+				propagatingTags: map[string]string{
+					"tracestate":   "dd=s:;o:rum:rum;t.dm:-4;t.usr.id:baz64~~,othervendor=t61rcWkgMzE",
+					"w3cTraceID":   "00000000000000001111111111111111",
+					"_dd.p.dm":     "-4",
+					"_dd.p.usr.id": "baz64==",
+				},
+			},
+		}
+
+		for _, test := range tests {
+			t.Run("", func(t *testing.T) {
+				// todo: replace that with env variable configuration
+				tracer := newTracer(WithPropagator(NewPropagator(&PropagatorConfig{}, &propagatorW3c{})))
+				assert := assert.New(t)
+				ctx, err := tracer.Extract(test.in)
+				assert.Nil(err)
+				sctx, ok := ctx.(*spanContext)
+				assert.True(ok)
+
+				assert.Equal(test.traceID, sctx.traceID)
+				assert.Equal(test.spanID, sctx.spanID)
+				assert.Equal(test.origin, sctx.origin)
+				p, ok := sctx.samplingPriority()
+				assert.True(ok)
+				assert.Equal(test.priority, p)
+
+				assert.Equal("00000000000000001111111111111111", sctx.trace.propagatingTags[w3cTraceIDTag])
+				assert.Equal(test.propagatingTags, sctx.trace.propagatingTags)
+
+			})
+		}
+	})
+
+	t.Run("extract/invalid", func(t *testing.T) {
```

Comment:
Maybe a few tests with some bad `tracestateHeaders` as well would be good

Reply:
Not ure what you mean. If you are referencing the changes you made in a different PR, I will create a branch where I can merge everything from our separate PRs and test it there

Reply:
Looking through the recent changes, this is the only one remaining, so mostly LGTM! Mainly I'm suggesting tests for things like this: https://github.com/DataDog/dd-trace-py/blob/a9ee02f0a671acd630ee0b79232be99372f05b2c/tests/tracer/test_context.py#L92
You're probably already covering a lot of cases like this, but essentially just making sure you have tracestate values that are invalid and will make sure that `fmt.SScanf` will fail as you expect, as well as bad span ids, trace ids, etc.